src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -218,13 +220,11 @@ private void validateUntaggedSummary(DetailNode ast) {
             log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
         }
         else if (!period.isEmpty()) {
-            final String firstSentence = getFirstSentence(ast);
-            final int endOfSentence = firstSentence.lastIndexOf(period);
-            if (!summaryDoc.contains(period)) {
+            final String firstSentence = getFirstSentence(ast, period);
+            if (!summaryDoc.contains(period) || firstSentence.isEmpty()) {
```

Comment:
It is confusing. It looks like the reason for the other way of getting the summary sentence is because that way reads the summary tag. I'm sure there's a way to clean this up, but it would make the PR bigger.

Reply:
ok, thanks for feedback, let not change it for now, may be we will see how to do this at the end.